File: android/gemcore/src/testFixtures/kotlin/com/gemwallet/android/testkit/WalletMock.kt
package com.gemwallet.android.testkit

import com.wallet.core.primitives.Account
import com.wallet.core.primitives.Chain
import com.wallet.core.primitives.Wallet
import com.wallet.core.primitives.WalletId
import com.wallet.core.primitives.WalletSource
import com.wallet.core.primitives.WalletType

fun mockWalletId(id: String = "wallet-1") = WalletId(id)

fun mockWallet(id: String = "wallet-1", name: String = "Wallet", type: WalletType = WalletType.Multicoin, accounts: List<Account> = emptyList(), source: WalletSource = WalletSource.Create) = Wallet(
    id = WalletId(id),
    name = name,
    index = 0,
    type = type,
    accounts = accounts,
    isPinned = false,
    source = source,
)

fun mockWalletMulticoin(address: String = "0xabc", name: String = "Main Wallet") = mockWallet(
    id = "multicoin_$address",
    name = name,
    accounts = listOf(mockAccount(chain = Chain.Ethereum, address = address)),
)

File: core/crates/primitives/src/testkit/wallet_mock.rs
use crate::{Account, Chain, Wallet, WalletId, WalletSource, WalletType};

impl Wallet {
    pub fn mock() -> Self {
        Self::mock_with_accounts(vec![Account::mock(Chain::Ethereum, "address")])
    }

    pub fn mock_with_chains(chains: &[Chain]) -> Self {
        Self::mock_with_accounts(Account::mock_chains(chains, "address"))
    }

    pub fn mock_with_type(wallet_type: WalletType, chains: &[Chain]) -> Self {
        Self {
            wallet_type,
            ..Self::mock_with_chains(chains)
        }
    }

    pub fn mock_with_id(id: WalletId, chains: &[Chain]) -> Self {
        Self {
            wallet_type: id.wallet_type(),
            id,
            ..Self::mock_with_chains(chains)
        }
    }

    pub fn mock_with_accounts(accounts: Vec<Account>) -> Self {
        Self {
            id: WalletId::Multicoin("0x1".to_string()),
            external_id: None,
            name: "wallet".to_string(),
            index: 0,
            wallet_type: WalletType::Multicoin,
            accounts,
            is_pinned: false,
            image_url: None,
            source: WalletSource::Import,
        }
    }
}
